rclcpp/include/rclcpp/create_client.hpp

```diff
@@ -44,15 +45,15 @@ create_client(
   std::shared_ptr<node_interfaces::NodeBaseInterface> node_base,
   std::shared_ptr<node_interfaces::NodeGraphInterface> node_graph,
   std::shared_ptr<node_interfaces::NodeServicesInterface> node_services,
+  std::shared_ptr<node_interfaces::NodeClockInterface> node_clock,
   const std::string & service_name,
-  const rclcpp::QoS & qos = rclcpp::ServicesQoS(),
-  rclcpp::CallbackGroup::SharedPtr group = nullptr)
+  const rclcpp::QoS & qos,
+  rclcpp::CallbackGroup::SharedPtr group,
+  bool enable_service_introspection)
```

Comment:
If we can, we should try and maintain backwards compatibility by leaving the old signature intact (and pass false for the service introspection flag and a nullptr for the clock interface.

Reply:
This method is supposed to be an internal method -- the only place outside of `rclcpp` it is used appears to be in `interactive_markers` and a PR has been opened to reconcile that in https://github.com/ihasdapie/interactive_markers/commit/f0e6419c4056f24ee02908728573492af86cd41f. 

That being said we can definitely another method to keep the old signature -- it would just mean that we would need to figure out what kind of behaviour we want if the clock is NULL -- does service introspection just not work or are we ok with no timestamp?